src/main/java/com/example/demo/dao/IPigeonsDAO.java

```diff
@@ -12,4 +12,8 @@ public interface IPigeonsDAO {
     void createPigeon(Pigeon pigeon);
 
     Optional<Pigeon> getPigeon(UUID id);
+
+    Optional<Pigeon> deletePigeon(UUID id);
+
+    Optional<Pigeon> updatePigeon(Pigeon newPigeon);
```

Comment:
Why would I create an entire class just to throw one exception of the same type?
What is this class for? Only for `throw new ResponseStatusException(HttpStatus.NOT_FOUND);`? Then I do not see a problem to use `ResponseStatusException`.
If Service and DAO will return void, then I do not understand how I should threw exception, because as I see we handle all exceptions in Controller.